dbuf-core/src/codegen/codegen.rs

```diff
@@ -0,0 +1,241 @@
+use core::panic;
+use std::iter;
+
+use crate::ast::elaborated::*;
+use pretty::BoxDoc;
+
+type Str = String; // Temporary
+const NEST_UNIT: isize = 4;
+
+#[derive(Clone)]
+struct NamedConstructor<Name, Constructor> {
+    name: Name,
+    constructor: Constructor,
+}
+
+struct RustType(String);
+
+// everything is needed to know about arbitrary field
```

Comment:
If you want this comment to appear in a generated documentation, then a triple slash (`///`) should be used to form a doc comment